extensions/wikia/CuratedContent/CuratedContentHelper.class.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * CuratedContent Helper
+ */
+class CuratedContentHelper {
+	const STR_ARTICLE = 'article';
+	const STR_BLOG = 'blog';
+	const STR_FILE = 'file';
+	const STR_CATEGORY = 'category';
+	const STR_EMPTY_CATEGORY = 'emptyCategory';
+	const STR_VIDEO = 'video';
+
+	public function processLogic( $sections ) {
+		$processedSections = [ ];
+		foreach ( $sections as $section ) {
+			$processedSections[ ] = $this->processLogicForSection( $section );
+		}
+		return $processedSections;
+	}
+
+	private function processLogicForSection( $section ) {
+		$section['image_id'] = (int)$section['image_id']; // fallback to 0 if it's not set in request
+
+		foreach ( $section['items'] as &$item ) {
+			$this->fillItemInfo( $item );
+		}
+
+		return $section;
+	}
+
+	private function fillItemInfo( &$item ) {
+		$title = Title::newFromText( $item['title'] );
+		if ( $title instanceof Title && $title->exists() ) {
+			$articleId = $title->getArticleId();
+			$namespaceId = $title->getNamespace();
+			$imageId = (int)$item['image_id'];
+
+			$item['type'] = self::getTypeFromNamespaceId( $namespaceId );
+
+			switch ( $item['type'] ) {
+				case self::STR_FILE:
+					$info = self::getVideoInfo( $title );
+					if ( !empty( $info ) ) {
+						$item['type'] = self::STR_VIDEO;
+						$item['video_info'] = $info;
+					}
+					break;
+
+				case self::STR_CATEGORY:
+					$category = Category::newFromTitle( $title );
+					if ( $category instanceof Category && $category->getID() ) {
+						$count = $category->getPageCount();
+						if ( empty( $count ) ) {
+							$item['type'] = self::STR_EMPTY_CATEGORY;
+						}
+					} else {
+						$item['type'] = null;
+					}
+					break;
+			}
+
+			if ( empty( $imageId ) ) {
+				$imageTitle = self::findFirstImageTitleFromArticle( $articleId );
+				if ( !empty( $imageTitle ) ) {
+					$imageId = $imageTitle->getArticleId();
```

Comment:
can getArticleId return null/0/false? If so what should we do? Do we need to handle this kind of situation?